### PR TITLE
add support for custom  object styles props for createThemedComponent

### PR DESCRIPTION
**Pull request by smsmdida**
… ei: containerStyle, indicatorStyle

### CODE REVIEW

**Comment by smsmdida** [comment]
@syaau any idea when this PR will get merged ?

File: lib/src/createThemedComponent.js [hunk 1]
```diff
@@ -41,32 +40,38 @@ export default function createThemedComponent(C, themedProps = []) {
...
     // eslint-disable-next-line camelcase, react/sort-comp
     UNSAFE_componentWillReceiveProps(nextProps) {
       const { theme, props } = this.state;
-
       let changed = false;
       // See if props have changed, only in which case change the state
       const newPropsState = themedProps.reduce((res, key) => {
-        const newValue = theme.getProp(nextProps[key]);
+        let newValue;
+        newValue = this.getThemedPropValue(nextProps[key], theme);
```
Comment (syaau) [inline lib/src/createThemedComponent.js]: @smsmdida if there isn't any reason to use `let` can we convert this to `const`.

Comment (syaau) [reply lib/src/createThemedComponent.js]: @smsmdida Also when you commit, can you also make sure the unnecessary changes due to whitespaces are removed, it just creates unnecessary git changes to look for.

File: lib/src/createThemedComponent.js [hunk 2]
```diff
@@ -41,32 +40,37 @@ export default function createThemedComponent(C, themedProps = []) {
...
         props: themedProps.reduce((res, key) => {
-          res[key] = theme.getProp(props[key]);
+          res[key] = this.getThemedPropValue(props[key], theme)
           return res;
         }, {}),
       };
     }
 
+    getThemedPropValue = (prop, theme) => {
+      if (typeof prop == 'object') {
```
Comment (syaau) [inline lib/src/createThemedComponent.js:50]: @smsmdida You are making a gross assumption that any prop of object type is a style, which is not correct. A style can be numeric value (the ones created with StyleSheet.create) and lots of prop values are object (even null is an object).

**Comment by syaau** [comment]
@smsmdida You are making a gross assumption that any prop of object type is a style, which is not correct. A style can be numeric value (the ones created with StyleSheet.create) and lots of prop values are object (even null is an object).

File: lib/src/createThemedComponent.js (hunk 1 above)
Comment (smsmdida) [reply lib/src/createThemedComponent.js]: @syaau no reason to use let, mistake when i refactored my code :D

**Comment by smsmdida** [comment]
@syaau we are checking on props mentioned in themedProps = [] not all props !

**Pull request merged**